lessons/dealing-with-big-data-and-network-analysis-using-neo4j.md

```diff
@@ -0,0 +1,453 @@
+
+
+You should see a network that looks something like this.
+
+{% include figure.html filename="new-neo4j-simple-match.png" caption="Neo4j results after a simple query" %}
+
+### A brief note on INDEX
+
+Creating an index is important for any database to run efficiently. 
+An index is a particular field in a database that is designated for the database to optimize so that lookups are as fast as possible.
+
+To create an index in Neo4j, we would issue the following Cypher command. 
+Creating an index only needs to be done once.
+```
+CREATE INDEX ON :COMPANY(id)
+```
+
+Create another index using the company name as well.
+```
+CREATE INDEX ON :COMPANY(name)
+```
+
+Creating this index will greatly speed up any queries we make based on the unique keys **id** and **name**.
+
+<div class="alert alert-warning">
+Don't create more indexes than you need. 
+Creating too many indexes will have the effect of slowing down your database.
+Again, designing your database so that you have a unique key to do lookups is crucial.
+</div>
+
+### Querying Neo4j: CREATE, MATCH, SET
+
+So far we have used the basic syntax of the Cypher query language. 
+We've seen that relationships on a graph are written quite intuitively using Cypher.
+```
+(n1:NODE)-[:relationship]->(n2:NODE)
+```
+In this section we used Cypher to CREATE entries in the database, MATCH existing data, and we used SET to alter existing data we found.
+
+> More on the [Cypher query language](https://neo4j.com/developer/cypher-query-language/) can be found on the Neo4j
+web site.
+
+# Putting it all together: A working example
+
+If we return to the [web interface on your local machine](http://localhost:7474) we can query our new database.
+Let's look at the firms that have the greatest number of connections (i.e. the highest degree).
+To calculate degree we can make a simple query with Cypher.
+
+```
+MATCH (c:COMPANY)
+set c.degree = size((c)-->());
+```
+
+This code simply matches to each node and counts the size (or degree) of each node.
+We use the SET command to set the degree value as an attribute of each node.
+
+Now we can examine those nodes with the highest degree. 
+Here we list companies where there are 75 or more connections (via high level employees or directors to other companies).
+```
+match (c0:COMPANY)-[r]-(c1) where c0.degree > 75
+return DISTINCT c0.name;
+```
+This results in the following companies:
+
+- TORONTO AND YORK RADIAL RAILWAY
+- MONTREAL ELECTRIC HEAT AND POWER
+- CANADIAN PACIFIC RAILWAY
+- TORONTO RAILWAY
+- DOMINION COAL
+- CANADIAN GENERAL ELECTRIC
+
+We can also try to examine all of the interlocks between these well-connected companies.
+
+```
+match (c0:COMPANY)-[r]-(c1) where c0.degree > 75 and c1.degree > 75
+return c0, r, c1;
+```
+
+{% include figure.html filename="graph_example.png" caption="Example graph" %}
+
+> You can download the data used in this lesson [here](http://jgmackay.com/) (search for the relevant blog posts).
```

Comment:
See comment about `alert-warning` above (although, this reads more like normal text that maybe doesn't need special formatting?)